AprilTagTrackers/Connection.cpp

```diff
@@ -78,14 +78,33 @@ void Connection::Connect()
 
 std::istringstream Connection::Send(std::string lpszWrite)
 {
+    int fd;
+
+    fd = open(lpszPipename, O_WRONLY);
+    write(fd, lpszWrite.c_str(), strlen(lpszWrite.c_str()) + 1);
+    close(fd);
+
+    fd = open(lpszPipename, O_RDONLY);
+    cbRead = read(fd, chReadBuf, BUFSIZE * sizeof(char));
+    close(fd);
+
+    std::cout << chReadBuf << std::endl;
+    chReadBuf[cbRead] = '\0';

```

Comment:
These lines should be swapped so that the string is terminated before being printed. This looks like it will be a bit spammy though.

Reply:
Honestly I didn't notice I put that output in there. I'll be sure to remove it.

Reply:
Actually I see I've just lazily copied that from Ju1ce's code...